llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py

```diff
@@ -869,6 +875,18 @@ async def adelete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
     def _create_odata_filter(self, metadata_filters: MetadataFilters) -> str:
         """Generate an OData filter string using supplied metadata filters."""
         odata_filter: List[str] = []
+
+        for f in metadata_filters.filters:
+            # Join values with ' or ' to create an OR condition inside the any function
+            value_str = " or ".join(
+                [
+                    f"t eq '{value}'" if isinstance(value, str) else f"t eq {value}"
+                    for value in f.value
+                ]
+            )
+            # Construct the filter query using the any function
+            return f"allowed_users_ids/any(t: {value_str})"
+
         for f in metadata_filters.legacy_filters():
             if not isinstance(f, ExactMatchFilter):
                 raise NotImplementedError(
```

Comment:
Should we remove this block? We either handle only ExactMatchFilters or we handle all filters

Above, it looks like you are still only handling exact match filters?